app/logic/Mail/ActiveSync/ActiveSyncAccount.ts

```diff
@@ -222,7 +233,14 @@ export class ActiveSyncAccount extends MailAccount {
       options.headers["X-MS-PolicyKey"] = await this.policyKey;
     }
     let wbxml = await request2WBXML({ [aCommand]: aRequest });
-    let response = await appGlobal.remoteApp.postHTTP(String(url), wbxml, "arrayBuffer", options);
+    let connection = appGlobal.remoteApp.postHTTP(String(url), wbxml, "arrayBuffer", options);
+    this.connections.add(connection);
+    let response;
+    try {
+      response = await connection;
+    } finally {
+      this.connections.remove(connection);
+    }
```

Comment:
This should be in a class `Semaphore`. I've implemented that.
@NeilRashbrook Could you please review my new code on master?

Reply:
The code is fine, but the comment looks as if it was copied and pasted...